pkg/vmtest/integration.go

```diff
@@ -301,3 +245,92 @@ func QEMU(o *Options) (*qemu.Options, error) {
 
 	return &o.QEMUOpts, nil
 }
+
+// ChooseTestInitramfs chooses which initramfs will be used for a given test and
+// places it at the location given by outputFile.
+// Default to the override initramfs if one was given. Else, build an initramfs
+// with the given parameters. If no uinit was provided, the generic one is used.
+func ChooseTestInitramfs(o uroot.Opts, uinit, outputFile string) error {
+	override := os.Getenv("UROOT_INITRAMFS")
+	if len(override) > 0 {
+		log.Printf("Overriding with initramfs %s", override)
+		if err := cp.Copy(override, outputFile); err != nil {
+			return err
+		}
+		return nil
+	}
+
+	if len(uinit) == 0 {
+		log.Printf("Defaulting to generic initramfs")
+		uinit = "github.com/u-root/u-root/integration/testcmd/generic/uinit"
+	}
+
+	_, err := CreateTestInitramfs(o, uinit, outputFile)
+	return err
+}
+
+// CreateTestInitramfs creates an initramfs with the given build options and
+// uinit, and writes it to the given output file. If no output file is provided,
+// one will be created, and it is the caller's responsibility to remove it.
+// The output file name is returned.
+func CreateTestInitramfs(o uroot.Opts, uinit, outputFile string) (string, error) {
+	// TODO need to bring back the DontSetEnv check?
```

Comment:
What was the original purpose of `DontSetEnv`?

Reply:
I think the purpose was to allow the user to supply BuildOpts.Env because we can't tell if o.Env is unset, so DontSetEnv was a way to let us know that we should leave it as is. There was a TODO about making that a pointer so we can easily make that check and don't need to do it in this roundabout way. I can do that in a follow up commit, just removing that option for now